src/Uno.UI/UI/Xaml/Controls/Layouter/Layouter.cs

```diff
@@ -255,6 +255,17 @@ public void Arrange(Rect finalRect)
 					}
 				}
 
+				// Alignment==Stretch --> arrange at the slot size minus margins
+				// Alignment!=Stretch --> arrange at the unclippedDesiredSize
+				if (Panel.HorizontalAlignment != HorizontalAlignment.Stretch)
```

Comment:
Original code from the public WinUI sources:

![image](https://github.com/unoplatform/uno/assets/31348972/4ad1f1dc-509d-4bf7-83c0-43e57183c02c)

Reply:
We also already do that for Skia/Wasm:

https://github.com/unoplatform/uno/blob/c0862855e0ed00489665fb1e7610c46eb118b78e/src/Uno.UI/UI/Xaml/FrameworkElement.Layout.crossruntime.cs#L403-L413